sdk/servicebus/service-bus/src/serviceBusMessage.ts

```diff
@@ -1055,8 +971,29 @@ export class ServiceBusMessageImpl implements ReceivedMessageWithLock {
    * @throws MessagingError if the service returns an error while renewing message lock.
    */
   async renewLock(): Promise<Date> {
-    this.throwIfMessageCannotBeSettled(this.getReceiverFromContext(), "renew the lock on");
-    return await this._context.managementClient!.renewLock(this.lockToken!);
+    if (this.sessionId) {
+      throw translate({
+        description: `Invalid operation on the message, message lock doesn't exist when dealing with sessions`,
+        condition: ErrorNameConditionMapper.InvalidOperationError
+      });
+    } else if (!this._context.requestResponseLockedMessages.has(this.lockToken!)) {
```

Comment:
You can use `if` instead of `else if` because the previous if block throws an error. Same for the other `else if` you have a few lines below. Makes it much easier to read

Reply:
Kept the `else-if` as is since I'm now logging and throwing from a commonplace to satisfy this comment https://github.com/Azure/azure-sdk-for-js/pull/8340#discussion_r417191202